sdk/core/Azure.Core/tests/TestFramework/RecordMatcher.cs

```diff
@@ -84,7 +98,11 @@ public virtual RecordEntry FindMatch(Request request, IList<RecordEntry> entries
                     score++;
                 }
 
+                //we only check Uri + RequestMethod for track1 record
+                if (!isTrack1Record)
+                {
                 score += CompareHeaderDictionaries(headers, entry.Request.Headers, ExcludeHeaders);
```

Comment:
This makes recordings way less useful, validating neither body nor headers.

Reply:
This compromise is not ideal, this is best I could figure out, please let me know if any other good suggestion.